Content/ContentTypeResolver/SingleSnippetSelectionResolver.php

```diff
@@ -60,56 +38,20 @@ public static function getContentType(): string
      */
     public function resolve($data, PropertyInterface $property, string $locale, array $attributes = []): ContentView
     {
-        /** @var StructureBridge $structure */
-        $structure = $property->getStructure();
-        $webspaceKey = $structure->getWebspaceKey();
-        $shadowLocale = $structure->getIsShadow() ? $structure->getShadowBaseLanguage() : null;
-
-        $params = $property->getParams();
-        /** @var bool $loadExcerpt */
-        $loadExcerpt = isset($params['loadExcerpt']) ? $params['loadExcerpt']->getValue() : false;
-        /** @var string $defaultArea */
-        $defaultArea = isset($params['default']) ? $params['default']->getValue() : null;
-
-        $snippetId = $data ?? null;
-        if (empty($snippetId) && $defaultArea) {
-            $defaultSnippetId = $this->getDefaultSnippetId($webspaceKey, $defaultArea, $locale);
-            $snippetId = $defaultSnippetId ?: null;
-        }
-
-        if (empty($snippetId)) {
-            return new ContentView(null, ['id' => null]);
-        }
-
-        /** @var SnippetBridge $snippet */
-        $snippet = $this->contentMapper->load($snippetId, $webspaceKey, $locale);
-
-        if (!$snippet->getHasTranslation() && null !== $shadowLocale) {
-            /** @var SnippetBridge $snippet */
-            $snippet = $this->contentMapper->load($snippetId, $webspaceKey, $shadowLocale);
-        }
-
-        $snippet->setIsShadow(null !== $shadowLocale);
-        $snippet->setShadowBaseLanguage($shadowLocale);
-
-        $resolvedSnippet = $this->structureResolver->resolve($snippet, $locale, $loadExcerpt);
-
-        return new ContentView($resolvedSnippet, ['id' => $data]);
-    }
-
-    private function getDefaultSnippetId(string $webspaceKey, string $snippetArea, string $locale): ?string
-    {
-        try {
-            /** @var SnippetDocument|null $snippet */
-            $snippet = $this->defaultSnippetManager->load($webspaceKey, $snippetArea, $locale);
-        } catch (WrongSnippetTypeException $exception) {
-            return null;
-        }
-
-        if (!$snippet) {
-            return null;
-        }
-
-        return $snippet->getUuid();
+        $snippetId = $data ?: null;
+
+        $contentView = $this->snippetSelectionResolver->resolve(
+            $snippetId ? [$snippetId] : null,
+            $property,
+            $locale,
+            $attributes
+        );
+        $content = $contentView->getContent();
+        $view = $contentView->getView();
+
+        return new ContentView(
+            $content[0] ?? null,
+            ['id' => $view['ids'][0] ?? null]
+        );
```

Comment:
is there a reason for implementing this different (eg. no early return) to the `SingleMediaSelectionResolver` and `SinglePageSelectionResolver`? 
guess it will work fine, just curious 🙂

Reply:
Yes, because if using an early return, if data is null, the default snippet won't be taken into account, because that handling is only in the `SnippetSelectionResolver` now

Reply:
makes sense - thanks!